src/na/CMakeLists.txt

```diff
@@ -171,6 +173,17 @@ if(NA_USE_UCX)
     ${NA_EXT_LIB_DEPENDENCIES}
     ${UCX_LIBRARIES}
   )
+  # Additional cmake testing
+  set(CMAKE_REQUIRED_INCLUDES ${UCX_INCLUDE_DIR})
+  set(CMAKE_REQUIRED_LIBRARIES ${UCX_LIBRARIES})
+
+  # Detect ucp_lib_query
+  check_symbol_exists(ucp_lib_query ucp/api/ucp.h NA_UCX_HAS_LIB_QUERY)
+  # Detect whether UCX has thread mode names
+  check_variable_exists(ucs_thread_mode_names NA_UCX_HAS_THREAD_MODE_NAMES)
```

Comment:
Is `ucs_thread_mode_names` a documented part of the API? Something new since 1.10?

Reply:
it's in master yes